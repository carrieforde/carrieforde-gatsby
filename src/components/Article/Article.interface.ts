import { Node } from "interfaces/node.interface";

export type ArticleProps = Node;
