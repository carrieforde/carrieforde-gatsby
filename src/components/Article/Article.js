import React from 'react';
import { Link } from 'gatsby';
import TimeStamp from '../TimeStamp/TimeStamp';

import articleStyles from './article.module.css';
import timeStampStyles from '../TimeStamp/timeStamp.module.css';

const Article = data => {
  const { excerpt, frontmatter, timeToRead, fields } = data.node,
    { title, date, category, description } = frontmatter;

  return (
    <article className={articleStyles.article}>
      <header className={articleStyles.articleHeader}>
        <span className={articleStyles.articleCategory}>{category}</span>
        <Link to={fields.slug}>
          <h2
            className={articleStyles.articleTitle}
            dangerouslySetInnerHTML={{ __html: title }}
          />
        </Link>
        <TimeStamp date={date} className={timeStampStyles.timeStampIsSmall} />
        <span className={articleStyles.articleSeparator}>&#9656;</span>
        <span className={articleStyles.articleTimeToRead}>
          {timeToRead} minute read
        </span>
      </header>
      <div className={articleStyles.articleDescription}>
        {description || excerpt}
      </div>
    </article>
  );
};

export default Article;
