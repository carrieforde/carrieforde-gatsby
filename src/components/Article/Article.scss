.article__title {
	font-weight: 300;
}

.article__category {
	color: map-get($primary, 8);
	display: block;
	font-size: $type-12;
	font-weight: 300;
	margin-bottom: $size-1;
	text-transform: uppercase;
}

.article__date,
.article__time-to-read,
.article__separator {
	color: map-get($neutral, 6);
	display: inline-block;
	font-size: $type-12;
	font-weight: 300;
	margin-bottom: $size-2;
	margin-top: $size-2;
}

.article__separator {
	margin-left: $size-1;
	margin-right: $size-1;
}
