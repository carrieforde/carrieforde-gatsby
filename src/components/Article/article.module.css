.articleTitle {
	font-weight: 300;
}

.articleTitle > * {
	background-color: transparent !important;
	font-size: inherit !important;
	padding: 0 !important;
}

.articleCategory {
	color: var(--color-primary-8);
	display: block;
	font-size: var(--font-size-1);
	font-weight: 300;
	margin-bottom: var(--size-1);
	text-transform: uppercase;
}

.articleDate,
.articleTimeToRead,
.articleSeparator {
	color: var(--color-neutral-6);
	display: inline-block;
	font-size: var(--font-size-1);
	font-weight: 300;
	margin-bottom: var(--size-2);
	margin-top: var(--size-2);
}

.articleSeparator {
	margin-left: var(--size-1);
	margin-right: var(--size-1);
}
