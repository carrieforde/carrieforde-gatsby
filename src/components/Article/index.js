import React from 'react';
import { Link } from 'gatsby';
import PropTypes from 'prop-types';
import TimeStamp from '../TimeStamp';

import styles from './article.module.css';
import timeStampStyles from '../TimeStamp/timeStamp.module.css';
import Category from '../Category';

const Article = (data) => {
  const { excerpt, frontmatter, timeToRead, fields } = data.node,
    { title, date, category, description } = frontmatter;

  return (
    <article className={styles.article}>
      <header className={styles.articleHeader}>
        <Category category={category} />
        <Link to={fields.slug}>
          <h2
            className={styles.articleTitle}
            dangerouslySetInnerHTML={{ __html: title }}
          />
        </Link>
        <TimeStamp date={date} className={timeStampStyles.timeStampIsSmall} />
        <span className={styles.articleSeparator}>&#9656;</span>
        <span className={styles.articleTimeToRead}>
          {timeToRead} minute read
        </span>
      </header>
      <div
        className={styles.articleDescription}
        dangerouslySetInnerHTML={{ __html: description || excerpt }}
      />
    </article>
  );
};

Article.propTypes = {
  data: PropTypes.object,
};

export default Article;
