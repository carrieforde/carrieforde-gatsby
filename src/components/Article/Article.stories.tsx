import React from 'react';
import { Story, Meta } from '@storybook/react';
import Article from './Article';
import { ArticleProps } from './Article.interface';

export default {
  title: 'Article',
  component: Article,
} as Meta;

const Template: Story<ArticleProps> = (args) => <Article {...args} />;

export const DefaultArticle = Template.bind({});

DefaultArticle.args = {
  frontmatter: {
    title: 'Inject dynamic data into static content with merge fields',
    date: 'March 08, 2021',
    category: 'JavaScript',
    description: null,
  },
  timeToRead: 9,
  excerpt:
    'Last year, I worked on upgrading one of our consumer-facing applications from a standard Angular application with static content hard-codedâ€¦',
  fields: {
    slug: '/dynamic-data-merge-fields/',
  },
};
