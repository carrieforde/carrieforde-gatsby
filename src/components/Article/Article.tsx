import { Link } from 'gatsby';
import React from 'react';
import ReactHtmlParser from 'react-html-parser';
import Category from '../Category/Category';
import TimeStamp from '../TimeStamp/TimeStamp';
import { ArticleProps } from './Article.interface';
import {
  article,
  articleDescription,
  articleHeader,
  articleSeparator,
  articleTimeToRead,
  articleTitle,
} from './article.module.css';

const Article: React.FC<ArticleProps> = ({
  excerpt,
  frontmatter,
  timeToRead,
  fields,
}) => {
  const { title, date, category, description } = frontmatter;

  return (
    <article className={article}>
      <header className={articleHeader}>
        <Category category={category} />
        <Link to={fields.slug}>
          <h2 className={articleTitle}>{ReactHtmlParser(title)}</h2>
        </Link>
        <TimeStamp date={date} isSmall={true} />
        <span className={articleSeparator}>&#9656;</span>
        <span className={articleTimeToRead}>{timeToRead} minute read</span>
      </header>
      <div className={articleDescription}>
        {ReactHtmlParser(description ?? excerpt)}
      </div>
    </article>
  );
};

export default Article;
