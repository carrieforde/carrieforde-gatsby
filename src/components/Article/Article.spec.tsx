import { render } from '@testing-library/react';
import React from 'react';
import { DefaultArticle } from './Article.stories';

describe('Article', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<DefaultArticle {...DefaultArticle.args} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
