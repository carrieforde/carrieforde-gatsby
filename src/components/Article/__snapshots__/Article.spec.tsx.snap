// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article should render correctly 1`] = `
<DocumentFragment>
  <article
    class="article"
  >
    <header
      class="articleHeader"
    >
      <a
        class="category"
        href="/category/javascript"
      >
        JavaScript
      </a>
      <a
        href="/dynamic-data-merge-fields/"
      >
        <h2
          class="articleTitle"
        >
          Inject dynamic data into static content with merge fields
        </h2>
      </a>
      <time
        class="timeStamp timeStampIsSmall"
        datetime="2021-03-08T08:00:00.000Z"
      >
        March 08, 2021
      </time>
      <span
        class="articleSeparator"
      >
        ▸
      </span>
      <span
        class="articleTimeToRead"
      >
        9 minute read
      </span>
    </header>
    <div
      class="articleDescription"
    >
      Last year, I worked on upgrading one of our consumer-facing applications from a standard Angular application with static content hard-coded…
    </div>
  </article>
</DocumentFragment>
`;
